### PR TITLE
[11.0][mig] privacy consent

### PR DESCRIPTION
**Pull request by fkantelberg**
Hello everyone,
this is the port of the privacy_consent module. Due to changes in v11 _ir.actions.server_ odoo run into an infinite loop which required the changes in the xml.

### CODE REVIEW

**Review by chienandalu — CHANGES_REQUESTED** [review]
Thanks @fkantelberg Can you get the latest commits fixing some issues in `10.0`?: https://github.com/OCA/data-protection/commits/10.0/privacy_consent

@Yajo Can you take a look on this?

File: privacy_consent/data/ir_actions_server.xml [hunk 1]
```diff
@@ -8,16 +9,9 @@
...
-        <field name="write_expression">record.partner_id</field>
-    </record>
-
-    <record id="update_opt_out_line_1" model="ir.server.object.lines">
-        <field name="server_id" ref="update_opt_out"/>
-        <field name="col1" ref="mail.field_res_partner_opt_out"/>
-        <field name="type">equation</field>
-        <field name="value">not record.accepted</field>
+        <field name="state">code</field>
+        <field name="code">for record in records:
```
Comment (yajo) [inline privacy_consent/data/ir_actions_server.xml]:
```suggestion
        <field name="code">records.mapped("partner_id").write({'opt_out': not record.accepted})</field>
```

File: privacy_consent/data/ir_actions_server.xml [hunk 2]
```diff
@@ -8,16 +9,9 @@
...
+            record.partner_id.write({'opt_out': not record.accepted})</field>
```
Comment (yajo) [inline privacy_consent/data/ir_actions_server.xml]:
```suggestion
```

**Review by yajo — CHANGES_REQUESTED** [review]
It looks good, overall. 2 requirements, though:

1. Include latest commits from branch 10.0. They are very important fixes.
1. Squash your migration commits in 1.

Also a suggestion to make it look better:

**Comment by fkantelberg** [comment]
@Yajo @chienandalu Thank you for your feedback. I did your wanted changes.

File: privacy_consent/data/ir_cron.xml [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
      License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
 
 <data>
 
     <record id="cron_auto_consent" model="ir.cron">
         <field name="name">Request automatic data processing consents</field>
-        <field name="model">privacy.activity</field>
-        <field name="function">_cron_new_consents</field>
+        <field name="model_id" ref="model_privacy_activity"/>
+        <field name="code">model._cron_new_consents()</field>
```
Comment (chienandalu) [inline privacy_consent/data/ir_cron.xml:12]: You've missed the `state` key here
```suggestion
        <field name="state">code</field>
        <field name="code">model._cron_new_consents()</field>
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
Thanks @fkantelberg 

There's a little issue in the `privacy` module form view of the activity with the domain, as the widget used should be `domain` and not `char_domain` which is a deprecated key. It's out of the scope for this module, but if want to make the fix I note it here :)

I little remark on your code below:

**Comment by fkantelberg** [comment]
@chienandalu I added the state and made a second commit for the view change.

File: privacy_consent/views/res_partner.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        <field name="name">Add consent smart button</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field
+            name="groups_id"
+            eval="[(4, ref('privacy.group_data_protection_user'))]"
+        />
+        <field name="arch" type="xml">
+            <div name="button_box" position="inside">
+                <!-- TODO Change icon to fa-handshake-o in Odoo 11 -->
```
Comment (yajo) [inline privacy_consent/views/res_partner.xml]: Let's do this now.

```suggestion
```

File: privacy_consent/views/res_partner.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                <button
+                    attrs='{"invisible": [("privacy_consent_count", "=", 0)]}'
+                    class="oe_stat_button"
+                    context='{"search_default_partner_id": active_id}'
+                    icon="fa-gavel"
```
Comment (yajo) [inline privacy_consent/views/res_partner.xml]:
```suggestion
                    icon="fa-handshake-o"
```

File: privacy_consent/views/privacy_activity.xml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+<data>
+
+    <record id="activity_form" model="ir.ui.view">
+        <field name="name">Add consent fields</field>
+        <field name="model">privacy.activity</field>
+        <field name="inherit_id" ref="privacy.activity_form"/>
+        <field name="arch" type="xml">
+            <div name="button_box" position="inside">
+                <!-- TODO Change icon to fa-handshake-o in Odoo 11 -->
```
Comment (yajo) [inline privacy_consent/views/privacy_activity.xml]: Let's obey this comment.
```suggestion
```

File: privacy_consent/views/privacy_activity.xml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                <button
+                    attrs='{"invisible": [("consent_required", "=", False)]}'
+                    class="oe_stat_button"
+                    context='{"search_default_activity_id": active_id}'
+                    icon="fa-gavel"
```
Comment (yajo) [inline privacy_consent/views/privacy_activity.xml]:
```suggestion
                    icon="fa-handshake-o"
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by fkantelberg** [comment]
@Yajo I haven't seen that comment in my tests but thanks for noticing. I changed the icons

**Review by yajo — APPROVED** [review]
/ocabot merge

**Pull request merged**